### PR TITLE
BUGFIX: updating tests (PR to 0.8.x)

### PR DESCRIPTION
**Pull request by to266**
Just like https://github.com/hyperspy/hyperspy/pull/959, but for the 0.8.x version

...By mistake I started with the master version, so this is cherry-picked from the #959. As a result, I may have accidentally picked up a new feature or two. Hopefully I did not, but better for someone else checks too.

### CODE REVIEW

File: hyperspy/tests/signal/test_fancy_indexing.py [hunk 1]
```diff
@@ -33,80 +33,65 @@ def setUp(self):
         self.data = self.signal.data.copy()
 
     def test_slice_None(self):
-        s = self.signal[:]
+        s = self.signal.isig[:]
```
Comment (dnjohnstone) [inline hyperspy/tests/signal/test_fancy_indexing.py]: Is this as it should be in 0.8.5? I forgot which version this is deprecated in?

File: hyperspy/tests/signal/test_fancy_indexing.py [hunk 2]
```diff
@@ -243,19 +237,10 @@ def setUp(self):
         self.signal.axes_manager._axes[1].navigate = True
         self.signal.axes_manager._axes[2].navigate = False
```
Comment (dnjohnstone) [inline hyperspy/tests/signal/test_fancy_indexing.py]: Did the test below disappear for a reason?

File: hyperspy/tests/signal/test_fancy_indexing.py [hunk 3]
```diff
@@ -327,24 +312,20 @@ def test_negative_float_step(self):
 class TestEllipsis:
 
     def setUp(self):
-        self.signal = Signal(np.arange(2 ** 4).reshape(
-            (2, 2, 2, 2)))
+        self.signal = Signal(np.arange(2 ** 5).reshape(
+            (2, 2, 2, 2, 2)))
         self.data = self.signal.data.copy()
```
Comment (dnjohnstone) [inline hyperspy/tests/signal/test_fancy_indexing.py]: Did the test below disappear for a reason?

**Comment by dnjohnstone** [comment]
There are two tests that seem to have disappeared in this PR - which I don't think should have... Other than that, I think this is ready to merge. I think the cherry picking went ok.

**Comment by to266** [comment]
In this case I took the liberty to remove all the "deprecated style" tests - even though the functions technically are still there, I don't think it's worth even testing that they work (in fact if they broke, it might be better, as that may encourage people to switch to the proper syntax).

Given the above thinking, the two removed tests were not necessary.

**Comment by dnjohnstone** [comment]
ok thanks for clarifying that - I think this is fine then, as far as I can see.

**Pull request merged**